wafw00f/main.py

```diff
@@ -89,7 +94,7 @@ def genericdetect(self):
             resp1 = self.performCheck(self.normalRequest)
             if 'User-Agent' in self.headers:
                 del self.headers['User-Agent']  # Deleting the user-agent key from object not dict.
-            resp3 = self.customRequest(headers=def_headers)
+            resp3 = self.customRequest(headers=self.merge_two_dicts(def_headers, self.headers))
```

Comment:
Merging the dicts ultimately defeats the purpose. But surprisingly you spotted an interested bug. we are using `def_headers`, while in reality we should be using `self.headers`.

Reply:
Resolved by 7851b94.